### ISSUE TITLE
fyne navigation menu issues

### ISSUE THREAD

**Issue by itswisdomagain** [raedahgroup/godcr-old#401]
- When the fyne interface is launched, all the functions for rendering the different pages get invoked instead of when the appropriate tab/menu item is selected. See the calls to render all pages as soon as fyne is started here: https://github.com/raedahgroup/godcr/blob/master/fyne/pages/menu.go#L67-L74.
Admittedly, all the pages are not rendered simultaneously, but are pre-loaded. Clicking on the tab menu item only displays the pre-loaded content. Quite obviously not a good idea to preload page contents.

- Visiting a page (say staking page), leaving to another page (say overview) and returning to the page previously visited does not cause the page to reload. There's a hack here https://github.com/raedahgroup/godcr/blob/master/fyne/pages/menu.go#L78-L91, which I've previously mentioned to be not okay - using a goroutine to refresh dynamic content on the active tab page every second. The recommended approach is to only trigger content reload when a tab menu item is activated/clicked.

A temporary fix to issue 2 above would be to track changes to the active tab index in that goroutine and only reload content if the tab index was actually changed; i.e. if I stay on tab 1 for 10 seconds, instead of reloading tab 1 content 10 times (for each second I remain on the tab), content should be reloaded only once.

Considering that it is not currently possible to load dynamic contents using fyne tab menus, I recommend changing the canvas item (view) used for the menu. Tabs require that the content for each page be loaded from the beginning and does not allow refreshing content when switching between pages.

An alternative to changing the canvas item (view) used for the menu would be to make changes to the `TabContainer` canvas object upstream or within godcr (extending `TabContainer` with additional helper methods perhaps) to account for the issues mentioned above.

**Comment by metaclips** [comment]
Tabcontainer contents are all loaded on start and are static, to dynamically change content "data", the go routine was used, we are not reloading the selected tab every seconds it's quite obvious that doing that wont give users interactivity but rather we are updating selected data in the page. I think fetching contents only when users change tabs doesnt just cut it, imagine changing tabs repeatedly to check for a new transaction, when it's pending and when it's confirmed, mobile wallet doesnt have that feature godcr shouldn't.

**Comment by itswisdomagain** [comment]
> Tabcontainer contents are all loaded on start and are static, to dynamically change content "data", the go routine was used, we are not reloading the selected tab every seconds it's quite obvious that doing that wont give users interactivity but rather we are updating selected data in the page.

My grammar was wrong, I agree that only some portion of the active page is being updated. My concern is with the frequency, refreshing/reloading some data on the active page every second is obviously a bad idea.

> I think fetching contents only when users change tabs doesnt just cut it, imagine changing tabs repeatedly to check for a new transaction, when it's pending and when it's confirmed, mobile wallet doesnt have that feature godcr shouldn't.

Mobile wallets do not use a repeating 1-second timer to check for transaction updates. There is a better way to implement that feature - using transaction notification listener. Until such a time when this is implemented, it's very fine to switch tabs to get updated information. Definitely better than refreshing said information every second.

We can have an issue created to implement tx notification listener. It's something that can be done in a pretty small PR in a pretty short timeframe when needed.

**Comment by metaclips** [comment]
Yes we both discussed about transaction notification listener function recently, tabcontainer doesn't support a calling function when a tab is changed, best thing to do is to implement a notification listener function instead. Would be lovely if you can create an issue on that instead.
